src/composable/conditionalorder.ts

```diff
@@ -0,0 +1,253 @@
+import { BigNumber, ethers, utils } from 'ethers'
+import { ContextFactory } from './multiplexer'
+import { keccak256 } from 'ethers/lib/utils'
+
+// Define the ABI tuple for the TWAPData struct
+export const CONDITIONAL_ORDER_PARAMS_ABI = ['tuple(address handler, bytes32 salt, bytes staticInput)']
+
+export type ConditionalOrderParams = {
+  readonly handler: string
+  readonly salt: string
+  readonly staticInput: string
+}
+
+/**
+ * An abstract base class from which all conditional orders should inherit.
+ *
+ * This class provides some basic functionality to help with handling conditional orders,
+ * such as:
+ * - Validating the conditional order
+ * - Creating a human-readable string representation of the conditional order
+ * - Serializing the conditional order for use with the `IConditionalOrder` struct
+ * - Getting any dependencies for the conditional order
+ * - Getting the off-chain input for the conditional order
+ *
+ * **NOTE**: Instances of conditional orders have an `id` property that is a `keccak256` hash of
+ *           the serialized conditional order.
+ */
+export abstract class BaseConditionalOrder<T, P> {
```

Comment:
Would be great to add one more method `createWithContext`.
Example: https://github.com/cowprotocol/cowswap/blob/develop/src/modules/twap/services/createTwapOrderTxs.ts#L29

It will be very useful for creating callData

Reply:
And `remove` :)
https://github.com/cowprotocol/cowswap/blob/develop/src/modules/twap/services/cancelTwapOrderTxs.ts#L15

Reply:
+decoding 🙏 
https://github.com/cowprotocol/cowswap/blob/develop/src/modules/twap/services/fetchTwapOrdersFromSafe.ts#L108

Reply:
Most of these it was uncertain exactly which area of the API they were going to sit in (whether to go into the "multiplexer", or to go into into the `BaseConditionalOrder` class). It's an area where `ComposableCoW` kind of has a foot in two camps - one in the creation of "single orders", and another when creating "multiple orders" using Merkle Roots.

Now having taken some time to think a little more about it, it does seem that `BaseConditionalOrder` is the best fit from a consuming-API perspective. A long way of saying, your wish is my command! 🪄

Reply:
RE: `create` and `remove`, I've added calldata generators for these.

In regards to decoding, this is already covered by `deserialize` which each order type implements (will deserialiez from an ABI encoded `params` struct).